apps/req.c

```diff
@@ -765,8 +765,9 @@ int req_main(int argc, char **argv)
         }
     }
     if (newreq || gen_x509) {
-        if (pkey == NULL /* can happen only if !newreq */) {
-            BIO_printf(bio_err, "Must provide a signature key using -key\n");
+        if (CAcert == NULL && pkey == NULL /* can happen only if !newreq */) {
+            BIO_printf(bio_err, "Must provide a signature key using -key or"
+                " provide both -CA and -CAkey\n");
```

Comment:
Actually the CA key can be placed in the same file as the CA certificate so this is not completely correct.